api/v1alpha4/azurecluster_conversion.go

```diff
@@ -42,6 +42,54 @@ func (src *AzureCluster) ConvertTo(dstRaw conversion.Hub) error {
 	// Restore list of virtual network peerings
 	dst.Spec.NetworkSpec.Vnet.Peerings = restored.Spec.NetworkSpec.Vnet.Peerings
 
+	// Restore API Server LB IP tags.
+	for _, restoredFrontendIP := range restored.Spec.NetworkSpec.APIServerLB.FrontendIPs {
+		for _, dstFrontendIP := range dst.Spec.NetworkSpec.APIServerLB.FrontendIPs {
+			if restoredFrontendIP.Name == dstFrontendIP.Name && restoredFrontendIP.PublicIP != nil {
+				dstFrontendIP.PublicIP.IPTags = restoredFrontendIP.PublicIP.IPTags
+			}
+		}
+	}
+
+	// Restore outbound LB IP tags.
+	if restored.Spec.NetworkSpec.ControlPlaneOutboundLB != nil {
+		for _, restoredFrontendIP := range restored.Spec.NetworkSpec.ControlPlaneOutboundLB.FrontendIPs {
+			for _, dstFrontendIP := range dst.Spec.NetworkSpec.ControlPlaneOutboundLB.FrontendIPs {
+				if restoredFrontendIP.Name == dstFrontendIP.Name && restoredFrontendIP.PublicIP != nil {
+					dstFrontendIP.PublicIP.IPTags = restoredFrontendIP.PublicIP.IPTags
+				}
+			}
+		}
+	}
+	if restored.Spec.NetworkSpec.NodeOutboundLB != nil {
+		for _, restoredFrontendIP := range restored.Spec.NetworkSpec.NodeOutboundLB.FrontendIPs {
+			for _, dstFrontendIP := range dst.Spec.NetworkSpec.NodeOutboundLB.FrontendIPs {
+				if restoredFrontendIP.Name == dstFrontendIP.Name && restoredFrontendIP.PublicIP != nil {
+					dstFrontendIP.PublicIP.IPTags = restoredFrontendIP.PublicIP.IPTags
+				}
+			}
+		}
+	}
+
+	// Restore NAT Gateway IP tags.
+	for _, restoredSubnet := range restored.Spec.NetworkSpec.Subnets {
+		for i, dstSubnet := range dst.Spec.NetworkSpec.Subnets {
+			if dstSubnet.Name == restoredSubnet.Name {
+				dst.Spec.NetworkSpec.Subnets[i].NatGateway.NatGatewayIP.IPTags = restoredSubnet.NatGateway.NatGatewayIP.IPTags
```

Comment:
correct me if I'm wrong but my understanding is that since  dstSubnet is not a pointer it won't actually modify the original dst object if we assign `dstSubnet` directly, that's why we need the `i`

same as above